InternshipApp/InternshipApp/FlightsApp/View/Screens/FlightSelectionScreen.swift

```diff
@@ -0,0 +1,311 @@
+                                        Spacer()
+                                    }
+                                }
+                                .padding(.horizontal, 16)
+                                .padding(.vertical, 4)
+                                .frame(maxWidth: .infinity)
+                                .overlay(
+                                    RoundedRectangle(cornerRadius: 8)
+                                        .stroke(Color(.systemGray4), lineWidth: 1)
+                                )
+                                .padding(.horizontal)
+                            }
+                            Divider()
+                            HStack {
+                                Image(systemName: "airplane.departure")
+                                Text(flightViewModel.selectedFlightOffer?.airCompany ?? "")
+                                    .font(.title2)
+                                    .fontWeight(.light)
+                                
+                                Spacer()
+                                
+                                Text("$\(flightViewModel.selectedFlightOffer?.price ?? 0)")
+                                    .font(.title)
+                                    .fontWeight(.bold)
+                                    .foregroundStyle(Color(.lightblue))
+                            }
+                            .padding(.horizontal)
+                            
+                            HStack {
+                                VStack(alignment: .leading) {
+                                    Text("Promo code")
+                                        .font(.caption)
+                                        .foregroundStyle(.gray)
+                                    HStack {
+                                        VStack(alignment: .leading) {
+                                            Text("****")
+                                                .font(.caption)
+                                                .foregroundStyle(Color(.lightGray))
+                                        }
+                                        Spacer()
+                                    }
+                                }
+                                .padding(.horizontal, 16)
+                                .padding(.vertical, 10)
+                                .frame(maxWidth: .infinity)
+                                .overlay(
+                                    RoundedRectangle(cornerRadius: 8)
+                                        .stroke(Color(.systemGray4), lineWidth: 1)
+                                )
+                                .padding(.horizontal)
+                                
+                                ButtonView(title: "Book flight", style: .primary) {
+                                    if flightViewModel.validateFlightBooking() {
+                                        flightViewModel.navigateToOffers = true
```

Comment:
Can this be done inside a method? So we don't polute a View with all of this? :)